airbyte-integrations/connectors/source-zuora/source_zuora/zuora_client.py

```diff
@@ -0,0 +1,253 @@
+
+        status = None
+        success = "completed"
+        while status != success:
+            response = self.make_request(url=f"{self.url_base}/query/jobs/{dq_job_id}")
+            job_check = response.json()
+            status = job_check["data"]["queryStatus"]
+            if status in errors and cursor_error in job_check["data"]["errorMessage"]:
+                raise ZOQLQueryFieldCannotResolve
+            elif status in errors and obj_read_error in job_check["data"]["errorMessage"]:
+                raise ZOQLQueryCannotProcessObject
+            elif status in errors:
+                raise ZOQLQueryFailed(response)
+        return job_check
+
+    def get_job_data(self, response: requests.Response) -> List:
+        """
+        Get actual output from submited data query job, when it's completed, using the link inside status job response
+        """
+
+        job_data = requests.get(response["data"]["dataFile"])
+        job_data.raise_for_status()
+        data = [j.loads(line) for line in job_data.text.splitlines()]
+        return data or []
+
+    def get_data(
+        self, q_type: str = "select", obj: str = None, date_field: str = None, start_date: str = None, end_date: str = None
+    ) -> Iterable:
+        """
+        Convinient wraper for `Make Query > Submit > Check > Get` operations
+        """
+
+        query = self.make_query(q_type, obj, date_field, start_date, end_date)
+        submit = self.submit_job(query)
+        check = self.check_job_status(submit)
+        get = self.get_job_data(check)
+        yield from get
+
+    def get_data_with_date_slice(
+        self, q_type: str, obj: str, date_field: str, start_date: str, window_days: int, end_date: str = None
+    ) -> Iterable:
+        """
+        Wraper for get_data using date-slices as pages, to overcome:
+        rate limits, data-job size limits, data-job performance issues on the server side.
+        """
+
+        # Parsing input dates
+        start = pendulum.parse(start_date).astimezone()
+        # If there is no `end_date` as input - use now()
+        end = pendulum.parse(end_date).astimezone() if end_date else pendulum.now().astimezone()
+        # initiating slice_start/end for date slice
+        slice_start = start
+        # if slice_end is bigger than the input end_date, switch to the end_date
+        slice_end = min(start.add(days=window_days), end)
+        while slice_start < end:
+            yield from self.get_data(q_type, obj, date_field, self.to_datetime_str(slice_start), self.to_datetime_str(slice_end))
+            slice_start = slice_end
+            slice_end = min(slice_end.add(days=window_days), end)
+
+    @staticmethod
+    def to_datetime_str(date: datetime) -> str:
+        """
+        The output from this method should be formated as follows:
+        :: "YYYY-mm-dd HH:mm:ss Z"
+        :: example: '2021-07-15 07:45:55 -07:00'
+        """
+        return f"{date.to_datetime_string()} {date.strftime('%Z')}"
```

Comment:
maybe are we better off using the full obvious generation If you have already used the 'strftime' function?
return f"{date.strftime('%Y-%m-%d %H:%M:%S %Z')}"